### PR TITLE
changed calculation for twitch amplitudes

### PR DESCRIPTION
**Pull request by NikitaPoly**

### CODE REVIEW

File: src/pulse3D/metrics.py [hunk 1]
```diff
@@ -154,22 +153,28 @@ def calculate_amplitudes(
...
-            if rounded:
-                amplitude_value = round(amplitude_value, 0)
 
-            estimates_dict[iter_twitch_idx] = amplitude_value
+        for twitchPeakX, twitchData in coordinates.iterrows():
+            CONVERSION = 100000000
+            C10X = twitchData.loc["time", "contraction", 10] * CONVERSION
+            C10Y = twitchData.loc["force", "contraction", 10]
+            R90X = twitchData.loc["time", "relaxation", 90] * CONVERSION
+            R90Y = twitchData.loc["force", "relaxation", 90]
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py:173]: These variables should have lower case names

File: src/pulse3D/metrics.py [hunk 2]
```diff
@@ -154,22 +153,28 @@ def calculate_amplitudes(
...
+            slope = (R90Y - C10Y) / (C10X - R90X)
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py:174]: This should be `(r90y - c10y) / (r90x - c10x)`

File: src/pulse3D/metrics.py [hunk 3]
```diff
@@ -154,22 +153,28 @@ def calculate_amplitudes(
...
+            twitchBaseY = slope * twitchPeakX
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py:175]: This formula assumes that the line starts at `(0, 0)`, but it starts at `(c10x, c10y)`, so this should be `twitch_base_y = c10y + slope * (twitch_peak_x - c10x)`

File: src/pulse3D/metrics.py [hunk 4]
```diff
@@ -154,22 +153,28 @@ def calculate_amplitudes(
...
-
-            amplitude_value = (
-                (peak_amplitude - prior_amplitude) + (peak_amplitude - subsequent_amplitude)
-            ) / 2
-
-            if rounded:
-                amplitude_value = round(amplitude_value, 0)
 
-            estimates_dict[iter_twitch_idx] = amplitude_value
+        for twitchPeakX, twitchData in coordinates.iterrows():
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py:168]: snake_case

File: CHANGELOG.rst [hunk 5]
```diff
@@ -13,6 +13,10 @@ Added:
 
 - Ability to interpolate stimulator output waveforms from start timepoints of each subprotocol
 
+Changed:
```
Comment (tannermpeterson) [inline CHANGELOG.rst:16]: This PR fixes a bug, so the changelog entry should be under `Fixed`, not `Changed`

**Review by tannermpeterson — CHANGES_REQUESTED** [review]

File: src/pulse3D/metrics.py [hunk 6]
```diff
@@ -154,22 +153,28 @@ def calculate_amplitudes(
...
+            CONVERSION = 100000000
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py:169]: What exactly is this conversion value? It's a magic a number with a name that doesn't not explain, but I'm assuming it has something to do with converting between µs and the interpolated timepoints. If so, it should use the `MICRO_TO_BASE_CONVERSION`

File: src/pulse3D/metrics.py [hunk 7]
```diff
@@ -154,22 +154,27 @@ def calculate_amplitudes(
         Returns:
             Pandas Series of float values representing the amplitude of each twitch
         """
+        twitch_indices_hashable = HashableDataFrame(DataFrame.from_dict(twitch_indices).T)
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py:157]: We need this to be fast enough to run in live view but unfortunately initializing dataframes is quite slow, so we need to stick to just using regular dicts here

File: src/pulse3D/metrics.py [hunk 8]
```diff
@@ -154,22 +154,27 @@ def calculate_amplitudes(
...
-            if rounded:
-                amplitude_value = round(amplitude_value, 0)
 
-            estimates_dict[iter_twitch_idx] = amplitude_value
+        for twitch_peak_x, twitch_data in coordinates.iterrows():
+            c10x = twitch_data.loc["time", "contraction", 10]
+            c10y = twitch_data.loc["force", "contraction", 10]
+            r90x = twitch_data.loc["time", "relaxation", 90]
+            r90y = twitch_data.loc["force", "relaxation", 90]
+            slope = (r90y - c10y) / (c10x - r90x)
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py:174]: (r90y - c10y) / (r90x - c10x)

File: src/pulse3D/metrics.py (hunk 7 above)
Comment (NikitaPoly) [reply src/pulse3D/metrics.py:157]: `TwitchWidth.calculate_twitch_widths` requires a HashableDataframe to be passed as the `twitch_indices`. I can move it up to its parent function?

Comment (tannermpeterson) [reply src/pulse3D/metrics.py:157]: I say let's leave it for now and assess the performance in the MA Controller later

File: src/pulse3D/metrics.py (hunk 8 above)
Comment (tannermpeterson) [reply src/pulse3D/metrics.py:174]: https://www.cuemath.com/geometry/finding-slope-from-two-points/ c10 is the 1st point and r90 is the 2nd point

File: src/pulse3D/metrics.py [hunk 9]
```diff
@@ -171,8 +171,8 @@ def calculate_amplitudes(
             c10y = twitch_data.loc["force", "contraction", 10]
             r90x = twitch_data.loc["time", "relaxation", 90]
             r90y = twitch_data.loc["force", "relaxation", 90]
-            slope = (r90y - c10y) / (c10x - r90x)
-            twitch_base_y = c10y + slope * (twitch_peak_x - c10x)
```
Comment (tannermpeterson) [inline src/pulse3D/metrics.py]: This doesn't need to change

Comment (NikitaPoly) [reply src/pulse3D/metrics.py]: Sorry I removed it for testing and forget to add it back.

File: src/pulse3D/metrics.py (hunk 8 above)
Comment (NikitaPoly) [reply src/pulse3D/metrics.py]: I tested with them both way and the results don't change but I left it as `(r90y - c10y) / (r90x - c10x)`.

**Review by tannermpeterson — APPROVED** [review]

**Pull request merged**